compiler/rustc_middle/src/hir/map/mod.rs

```diff
@@ -983,7 +983,11 @@ impl<'hir> Map<'hir> {
             Node::AnonConst(constant) => self.body(constant.body).value.span,
             Node::Expr(expr) => expr.span,
             Node::Stmt(stmt) => stmt.span,
-            Node::PathSegment(seg) => seg.ident.span,
+            Node::PathSegment(seg) => {
+                let ident_span = seg.ident.span;
+                ident_span
+                    .with_hi(seg.args.map_or_else(|| ident_span.hi(), |args| args.span_ext.hi()))
+            }
```

Comment:
I would guess that the only thing which may impact the amount of `process_obligations` would be this.

We now use a different `span` from `seg.ident` which may mean that we don't dedup obligations as readily as they now have different spans? Can try to take a deeper look at this in the near future